### ISSUE TITLE
Flip the value of `--experimental_python_import_all_repositories` to false and remove the flag

### ISSUE THREAD

**Issue by lberki** [bazelbuild/bazel#2636]
Filing a bug here so that we don't forget to get rid of Yet Another Command Line Option.

**Comment by damienmg** [comment]
Ok I won't manage to fix everything for that change so I am rescheduling it for 0.6

**Comment by aj-michael** [comment]
cc @vsethia and @ahumesky 

FYI, this change will heavily affect the Android testing infrastructure (android_device) that is being open sourced at  https://github.com/google/android-testing-support-library. It is mostly in Python and has several external repositories that it imports from.

**Comment by brandjon** [comment]
This is an interesting situation, in that we know we want to ban the old behavior and we even already have a flag to do it, but the new behavior doesn't have a complete story yet. See #7067.

In the meantime, we might go ahead and rename this to `--incompatible_repo_roots_are_not_on_pythonpath` (inverting the meaning) and make it available as an incompatible change, even though flipping it is blocked on more feature work being done. :-/

**Comment by guillaumep** [comment]
I just ran into an issue running Bazel that I tracked down to this flag.
I am packaging the PY2 and PY3 version of an app together, allowing us to start testing the PY3 version on staging environments while still running the PY2 version in prod.

When this flag is set to true, the PY3 version has PY2 dependencies in its PYTHONPATH, causing issues starting the application (the exact issue being that PY2 deps have the enum34 package which causes issues when using Python 3).

So just a note that this flag can cause some very unexpected behaviours.

**Comment by person142** [comment]
The current behavior also causes strangeness in `rules_python`-see https://github.com/bazelbuild/rules_python/issues/414#issuecomment-789416360 e.g.

Would adding the `--incompatible_repo_roots_are_not_on_pythonpath` flag to disable the behavior as suggested by https://github.com/bazelbuild/bazel/issues/2636#issuecomment-474008322 be an acceptable way to get the ball rolling again here?

**Comment by rickeylev** [comment]
cc @rickeylev

**Comment by amir-f** [comment]
Any updates? I'm here because of [this issue](https://github.com/bazelbuild/bazel/issues/14640#issuecomment-1718227554)

**Comment by fmeum** [comment]
`--enable_bzlmod` currently relies on this flag value as it mangles the repository names to something you can't reliably import in Python.

**Comment by rickeylev** [comment]
The corresponding rules_python issue is here: https://github.com/bazel-contrib/rules_python/issues/2943

I'm not sure if/when rules_python will get around to flipping it. It may just end up waiting for the "migrate builtin flags to rule repos" work to be done.

**Comment by meisterT** [comment]
Closing this in favor of the rules_python issue
